File: Exercism Codes/JavaScript/rectangles.js
export const count = (grid) => {
  let rect = 0;
  for (let i = 0; i < grid.length; i++) {
    for (let j = 0; j < grid[i].length; j++) {
      if (grid[i][j] == '+') {
        for (let k = j + 1; k < grid[i].length; k++) {
          if (grid[i].substring(j, k + 1).match(/^\+[+-]*\+$/)) {
            for (let l = i + 1; l < grid.length; l++) {
              if (grid[l].substring(j, k + 1).match(/^[+|].*[+|]$/)) {
                if (grid[l].substring(j, k + 1).match(/^\+[+-]*\+$/)) rect++;
              }
              else break;
            }
          }
        }
      }
    }
  }
  return rect;
}

File: Exercism Codes/Python/rectangles.py
import re

ex = re.compile("^\\+[+-]*\\+$")
it = re.compile("^[+|].*[+|]$")


def rectangles(strings):
    count = 0
    for i in range(len(strings)):
        for j in range(len(strings[i])):
            if strings[i][j] == '+':
                for k in range(j + 1, len(strings[i])):
                    if ex.match(strings[i][j:k + 1]):
                        for l in range(i + 1, len(strings)):
                            if it.match(strings[l][j:k + 1]):
                                if ex.match(strings[l][j:k + 1]):
                                    count += 1
                            else:
                                break
    return count
